src/js/actions/type.js

```diff
@@ -240,20 +241,21 @@ define(function (require, exports) {
      * @param {Document} document
      * @param {Immutable.Iterable.<Layers>} layers
      * @param {Color} color
-     * @param {boolean=} coalesce Whether to coalesce this operation's history state
-     * @param {boolean=} ignoreAlpha Whether to ignore the alpha value of the
+     * @param {boolean=} options.coalesce Whether to coalesce this operation's history state
+     * @param {boolean=} options.ignoreAlpha Whether to ignore the alpha value of the
      *  given color and only update the opaque color value.
      * @return {Promise}
      */
-    var setColor = function (document, layers, color, coalesce, ignoreAlpha) {
+    var setColor = function (document, layers, color, options) {
         var layerIDs = collection.pluck(layers, "id"),
             layerRefs = layerIDs.map(textLayerLib.referenceBy.id).toArray(),
             normalizedColor = color.normalizeAlpha(),
             opaqueColor = normalizedColor.opaque(),
             playObject = textLayerLib.setColor(layerRefs, opaqueColor),
-            typeOptions = _getTypeOptions(document.id, strings.ACTIONS.SET_TYPE_COLOR, coalesce);
+            typeOptions = _.merge(options,
```

Comment:
We need to pass the transaction ID in the batchPlay options so the descriptor.batchPlay knows not to play the descriptor playObjects but instead append them to the transaction list.